src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
>And you ignore/forget that the size() here gives not the preferred size specified in .ui files.
Here it returns {100, 30} which is too small.

Oops... Really, I forgot it.
>or initialize it later...

Using pointer to **CachedSettingValue** breaks its goals (we can't use `var = m_settingValue` etc.).
But you can use it the following way:
```c++
, m_storeDialogSize(SETTINGS_KEY("Dimension"))
```
in constructor and then:
```c++
if (m_storeDialogSize.value().isValid())
    resize(m_storeDialogSize);
```